File: appjava/src/main/java/com/byoutline/kickmaterial/model/ProjectDetails.java
package com.byoutline.kickmaterial.model;

import org.parceler.Parcel;

import java.util.List;

@Parcel(Parcel.Serialization.FIELD)
public class ProjectDetails extends Project {

    public List<Reward> rewards;
    public int commentsCount;
    public int updatesCount;
    public ProjectVideo video;

    public String getVideoUrl() {
        if (video == null) {
            return "";
        }
        return video.base;
    }

    public String getAltVideoUrl() {
        if (video == null) {
            return "";
        }
        return video.webm;
    }
}

File: app/src/main/java/com/byoutline/kickmaterial/model/ProjectDetails.kt
package com.byoutline.kickmaterial.model

import paperparcel.PaperParcel

@PaperParcel
class ProjectDetails(
        val rewards: List<Reward>?,
        val commentsCount: Int,
        val updatesCount: Int,
        val video: ProjectVideo?) : Project() {

    val videoUrl: String
        get() = video?.base ?: ""


    val altVideoUrl: String
        get() = video?.webm ?: ""

    companion object {
        @JvmField
        val CREATOR = PaperParcelProjectDetails.CREATOR
    }
}
